core/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -273,11 +277,73 @@ OpenTelemetryRum build(ServiceManager serviceManager) {
         InitializationEvents initializationEvents = InitializationEvents.get();
         applyConfiguration(serviceManager, initializationEvents);
 
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+        BufferDelegatingLogExporter bufferDelegatingLogExporter =
+                new BufferDelegatingLogExporter();
+
+        BufferDelegatingSpanExporter bufferDelegatingSpanExporter =
+                new BufferDelegatingSpanExporter();
+
+        SessionManager sessionManager =
+                SessionManager.create(timeoutHandler, config.getSessionTimeout().toNanos());
+
+        OpenTelemetrySdk sdk =
+                OpenTelemetrySdk.builder()
+                        .setTracerProvider(
+                                buildTracerProvider(
+                                        sessionManager,
+                                        application,
+                                        bufferDelegatingSpanExporter))
+                        .setMeterProvider(buildMeterProvider(application))
+                        .setLoggerProvider(
+                                buildLoggerProvider(
+                                        application, bufferDelegatingLogExporter))
+                        .setPropagators(buildFinalPropagators())
+                        .build();
+
+        otelSdkReadyListeners.forEach(listener -> listener.accept(sdk));
+
+        SdkPreconfiguredRumBuilder delegate =
+                new SdkPreconfiguredRumBuilder(
+                        application,
+                        sdk,
+                        timeoutHandler,
+                        sessionManager,
+                        config.shouldDiscoverInstrumentations(),
+                        serviceManager);
+
+        executorService.execute(
+                () -> {
+                    initializeExporters(
+                            serviceManager,
+                            initializationEvents,
+                            bufferDelegatingSpanExporter,
+                            bufferDelegatingLogExporter);
+                });
+
+        executorService.shutdown();
+
+        instrumentations.forEach(delegate::addInstrumentation);
+
+        return delegate.build();
+    }
+
+    private void initializeExporters(
```

Comment:
Is there a risk that the file based exporters fail to initialize, and we just hoard events in memory with nowhere to go?

Reply:
I wonder if we should utilize the`try-catch` in L364 to shutdown the delegating exporters.